### PR TITLE
Fe ncnews votes

### PR DESCRIPTION
**Pull request by Alishow1111**

### CODE REVIEW

File: src/components/ArticlePage.jsx [hunk 1]
```diff
@@ -10,16 +12,34 @@ import Comments from './Comments';
...
+            setVoteCount(articleResponse.votes);
             setLoading(false)
         })
     }, [])
 
+    function incrementVote(){
+        setVote(true);
+        setDownVote(false);
+        setVoteCount(1);
+        patchVotes(article_id, 1);
```
Comment (2202Hannah) [inline src/components/ArticlePage.jsx]: Make sure you are setting your vote amount before you make your patch request. The vote amount itself should be stored in state and updated each time your incrementVote or decrementVote function is invoked. It looks like you are currently changing the vote amount directly in your render on line 58 which may explain some of the strange behaviour of your code.

File: src/components/ArticlePage.jsx [hunk 2]
```diff
@@ -10,16 +12,34 @@ import Comments from './Comments';
...
+    }
+
```
Comment (2202Hannah) [inline src/components/ArticlePage.jsx:52]: You should also make sure that you are handling the vote change with optimistic rendering. To recap optimistic rendering should:
- updated the vote state before you make the request to your backend api
- if the request to your api is unsuccessful and has an error you will need to revert the vote amount back to it's original value.

You can see more about optimistic rendering in the notes here - https://notes.northcoders.com/courses/js-front-end/ui-behaviours-and-optimistic-rendering#optimistic-rendering

File: src/components/ArticlePage.jsx [hunk 3]
```diff
@@ -35,6 +55,7 @@ function ArticlePage() {
             <h3 style={{paddingTop:20}}>Written by: {article.author}</h3>
             <p style={{paddingTop: 20}}>{article.body}</p>
 
+            <ThumbUpRoundedIcon color={vote ? "primary" : ""} onClick={incrementVote}/> {article.votes + voteCount}  <ThumbDownRoundedIcon color={downVote ? "error" : ""} onClick={decrementVote} />
```
Comment (2202Hannah) [inline src/components/ArticlePage.jsx]: You should avoid changing/manipulating your vote amount like this. I would suggest having [votes, setVotes] in state and invoking setVotes in your increase and decrease functions.

File: src/components/utils/api.js [hunk 4]
```diff
@@ -28,6 +28,18 @@ function fetchCommentsByArticle (article_id){
...
+}
+
+function patchVotes (article_id, count){
+    return axios.patch(`https://nc-news-2.onrender.com/api/articles/${article_id}`, 
+    {
+        inc_votes: count
+    })
+    .then((response) =>{
+        return response;
+    })
```
Comment (2202Hannah) [inline src/components/utils/api.js:41]: Nice work with your axios request 👍

**Review by 2202Hannah — CHANGES_REQUESTED** [review]
Hey Ali! Nice work so far on this ticket. 

A few points to consider:
- The vote change isn't consistent when I refresh the page - it looks like the amount of votes that are stored in you backend api is changing, but what is displayed on the page (in state) is different to what is happening with your patch request. 
- When I first upvote and then downvote the value goes from 1 to -1 which doesn't feel particularly intuitive. 
- When I refreshed the page and tried again, when I downvote is decrements by 2 votes. 
- The user should be notified if the patch request fails. Currently if I switch to offline and try and vote it looks from the users perspective like the vote has worked. I suggest adding an error message for this scenario. 

Please submit another pr for this ticket when you are ready :)

File: src/components/ArticlePage.jsx [hunk 5]
```diff
@@ -16,28 +16,53 @@ function ArticlePage() {
...
         .then((articleResponse) => {
             setArticle(articleResponse);
             setVoteCount(articleResponse.votes);
             setLoading(false)
+
+            const hasVoted = localStorage.getItem(`vote_${article_id}`);
+            if (hasVoted === 'up') {
+                setVote(true);
+            } else if (hasVoted === 'down') {
+                setDownVote(true);
```
Comment (nc-linda-guester) [inline src/components/ArticlePage.jsx:32]: Wow! Definitely didn't expect anyone to use localStorage, well done!

**Review by nc-linda-guester — APPROVED** [review]
Well done!!

**Pull request merged**